graphql-server/src/databases/database.resolver.ts

```diff
@@ -91,6 +115,29 @@ export class DatabaseResolver {
     return dbs;
   }
 
+  @Mutation(_returns => [String])
+  async databasesByConnection(
+    @Args() args: AddDatabaseConnectionArgs,
+  ): Promise<string[]> {
+    if (this.conn.getWorkbenchConfig()?.connectionUrl === args.connectionUrl) {
+      return this.databases();
+    }
+    const workbenchConfig = getWorkbenchConfigFromArgs(args);
+    const ds = getDataSource(workbenchConfig);
+    await ds.initialize();
+    const dbs = await ds.query("SHOW DATABASES");
+    await ds.destroy();
```

Comment:
I think you need to put the query and return in a `try` block and put this in a `finally` so that the connection gets closed if there's an error as well